### DISCUSSION TITLE
Only allow numbers in input fields

### DISCUSSION THREAD

**Discussion by c0d3x** [react-hook-form/react-hook-form#2910] [Q&A]
When using RHF, what is the best way of making the input fields accept only numbers between 0-9?

My input is uncontrolled and I want it to only accept numbers, because I do som multiplications on the inputs watch event.
When having multiple inputs I need to make a total amount that is just displayed for the user.

I use YUP validation, and there I set it to number. 
But the watch event remains a string until I submit the form, then the input value becomes a number(guess YUP magic or something). So to be able to do multiplications between multiple inputs by the watch event, I also need to convert the string to number in order to multiply the array of input amount fields and get the total amount in realtime.

Any ideas?

**Comment by bluebill1049** [comment]
mask input or 

`<input type="text" pattern="\d*" maxlength="4">`?
